File: 2017/adventofcode.py
from termcolor import colored

def read_input(day):
    puzzle_input = [line.rstrip('\n')
                    for line in open("day%02d_input.txt" % day)]
    if len(puzzle_input) == 1:
        puzzle_input = puzzle_input[0]
    return puzzle_input


def answer(part, correct, calculated):
    if correct == 0:
        color = 'yellow'
    elif calculated == correct:
        color = 'green'
    else:
        color = 'red'
    message = "Part %d Answer %s" % (part, calculated)
    print colored(message, color)

File: 2017/adventofcode/lib/adventofcode.ex
defmodule Adventofcode do
  @moduledoc """
  Documentation for Adventofcode.
  """

  @doc """
  Hello world.

  ## Examples

      iex> Adventofcode.hello
      :world

  """
  def hello do
    :world
  end

  @doc """
  """
  def read_input(day) do
    pd = day |> Integer.to_string() |> String.pad_leading(2, "0")
    fname = "../day#{pd}_input.txt"
    data = File.read!(fname)
    # lines = String.split data, ~r{\r|\n|\r\n}, trim: true
    lines = String.split(data, ["\r", "\n", "\r\n"], trim: true)

    # lines = File.stream!(fname) |> Stream.map( &(String.replace(&1, "\n", "")) ) |> Enum.to_list

    case lines do
      [a] -> a
      a -> a
    end
  end

  def answer(part, correct, calculated) do
    color =
      cond do
        correct == 0 -> :yellow
        correct == calculated -> :green
        true -> :red
      end

    IO.puts(IO.ANSI.format([color, "Part #{part} Answer #{calculated}"], true))
  end
end
